File: LinearSearch.cpp
#include<iostream>
using namespace std;
class LinearSearch
{
    public:
    int Linearsearch(int Arr[],int isize,int no)
    {
        int i=0;
        for(i=0;i<isize;i++)
        {
            if(Arr[i]==no)
            {
                break;
            }
        }
        if(i==isize)
        {
            return -1;
        }
        else
        {
            return i;
        }
    }
};
int main()
{
    int length=0,i=0,no=0;
    int *p=NULL;
    int iret=0;

    cout<<"Enter the length of Array:\n";
    cin>>length;

    p=new int[length];

    cout<<"Element are:\n";
    for(i=0;i<length;i++)
    {
        cout<<"Element :"<<i+1<<"\n";
        cin>>p[i];
    }

    cout<<"Enter number which you want to search\n";
    cin>>no;

    LinearSearch obj;
    iret=obj.Linearsearch(p,length,no);
    if(iret==-1)
    {
        
        cout<<"Number is not present\n";
    }
    else
    {
         cout<<"Number  is present at index:"<<iret<<"\n";
    }

    return 0;
}

File: LinearSearch.java
import java.util.*;
class ArrayX
{
    public int Arr[];
    public ArrayX(int size)
    {
        Arr=new int[size];
    }
    public void Accept()
    {
        Scanner sobj=new Scanner(System.in);
        int i=0;
        System.out.println("Enter element:");
        for(i=0;i<Arr.length;i++)
        {
            Arr[i]=sobj.nextInt();
        }
    }

    public void Display()
    {
        System.out.println("Data are:");
        for(int i=0;i<Arr.length;i++)
        {
            System.out.println(Arr[i]);
        }
    }
}
class linearSearch extends ArrayX
{
    public linearSearch(int no)
    {
        super(no);
    }

    public boolean search(int no)
    {
        int i=0;
        for(i=0;i<Arr.length;i++)
        {
            if(Arr[i]==no)
            {
                break;
            }
        }
        if(i==Arr.length)
        {
            return false;
        }
        else
        {
            return true;
        }
    }
}
class LinearSearch
{
    public static void main(String arg[])
    {
        Scanner sobj=new Scanner(System.in);
        int length=0;
        System.out.println("Enter size: ");
        length=sobj.nextInt();
        linearSearch obj=new linearSearch(length);
        obj.Accept();
        obj.Display();
        System.out.println("Enter number you want to search:");
        int no=sobj.nextInt();
        boolean bRet=false;
         bRet=obj.search(no);
         if(bRet==true)
         {
             System.out.println("Number is their");
         }
         else
         {
             System.out.println("Number is not their");
         }

    }
}
